arkouda/pdarrayclass.py

```diff
@@ -3075,6 +3078,75 @@ def sqrt(pda: pdarray, where: Union[bool, pdarray] = True) -> pdarray:
     return power(pda, 0.5, where)
 
 
+# there's no need for typechecking, % can handle that
+def mod(dividend, divisor) -> pdarray:
+    """
+    Returns the element-wise remainder of division.
+
+    Computes the remainder complementary to the floor_divide function.
+    It is equivalent to np.mod, the remainder has the same sign as the divisor.
+
+    Parameters
+    ----------
+    dividend
+        The array being acted on by the bases for the modular division.
+    divisor
+        The array that will be the bases for the modular division.
+
+    Returns
+    -------
+    pdarray
+        Returns an array that contains the element-wise remainder of division.
+    """
+    return dividend % divisor
+
+
+@typechecked
+def fmod(dividend: Union[pdarray, numeric_scalars], divisor: Union[pdarray, numeric_scalars]) -> pdarray:
+    """
+    Returns the element-wise remainder of division.
+
+    It is equivalent to np.fmod, the remainder has the same sign as the dividend.
+
+    Parameters
+    ----------
+    dividend : numeric scalars or pdarray
+        The array being acted on by the bases for the modular division.
+    divisor : numeric scalars or pdarray
+        The array that will be the bases for the modular division.
+
+    Returns
+    -------
+    pdarray
+        Returns an array that contains the element-wise remainder of division.
+    """
+    if not builtins.all(
+        isSupportedNumber(arg) or isinstance(arg, pdarray) for arg in [dividend, divisor]
```

Comment:
Same question here. I would expect the `@typechecked` to handle this.

Reply:
Captured in #2528